src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,54 +1,115 @@
-import React from 'react';
+import React, { useState } from 'react';
+import classNames from 'classnames';
 import './FindMovie.scss';
 
-// import { MovieCard } from '../MovieCard';
-
-export const FindMovie: React.FC = () => (
-  <>
-    <form className="find-movie">
-      <div className="field">
-        <label className="label" htmlFor="movie-title">
-          Movie title
-        </label>
-
-        <div className="control">
-          <input
-            type="text"
-            id="movie-title"
-            placeholder="Enter a title to search"
-            className="input is-danger"
-          />
-        </div>
+import { MovieCard } from '../MovieCard';
+import { getMovie } from '../../api';
 
-        <p className="help is-danger">
-          Can&apos;t find a movie with such a title
-        </p>
-      </div>
+type Props = {
+  movies: Movie[],
+  setMovies: any,
+};
+
+export const FindMovie: React.FC<Props> = ({ movies, setMovies }) => {
+  const [query, setQuery] = useState<string>('');
+  const [movie, setMovie] = useState<Movie | null>(null);
+  const [fetchError, setFetchError] = useState(false);
+
+  const handleInputChange = (
+    event: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    setQuery(event.target.value);
+  };
+
+  const resetingInputAndError = () => {
+    setFetchError(false);
+    setMovie(null);
+  };
+
+  const findMovie = async () => {
+    resetingInputAndError();
+    const movieFromServer = await getMovie(query);
+
+    if (movieFromServer.Error) {
+      await setFetchError(true);
+    } else {
+      await setMovie(movieFromServer);
+    }
+  };
 
-      <div className="field is-grouped">
-        <div className="control">
-          <button
-            type="button"
-            className="button is-light"
-          >
-            Find a movie
-          </button>
+  const handleAddMovie = () => {
+    if (!movies.some((movieFromList: Movie) => (
+      movieFromList.imdbID === movie?.imdbID
+    )) && movie) {
```

Comment:
no need. If movie will be null, the some method return false

```suggestion
    if (!movies.some((movieFromList: Movie) => (
      movieFromList.imdbID === movie?.imdbID
    ))) {
```